### PR TITLE
[NO-TICKET] Remove console.log, create eslint check for it

### PR DESCRIPTION
**Pull request by dmethvin-gov**
## Summary

While looking around the source, I found a stray `console.log` that is most likely leftover from recent debugging. This PR removes it and also adds an ESLint rule to fail use of any console method except console.warn and console.error in the production source code. It still allows logging in stories. 

Although this counts console.log as an error that will fail ESLint CI, it doesn't prevent compilation of log messages during local debugging. The goal is to prevent shipping code with debugging messages.

If this seems too complex, feel free to just remove the console.log and close this PR. 😸 

## How to test

Run the unit tests. You should no longer see the console message appear in the output.

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.
- [x] Selected appropriate release milestone

### If this is a change to design:

- [ ] If visual regression image references have been changed, design **MUST** be assigned to review. In this instance, **designer approval is a requirement** before the PR can be merged.

### If this is a change to code:

- [x] Created or updated unit tests to cover any new or modified code
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:update`)

### If this is a change to documentation:

- [ ] Checked for spelling and grammatical errors

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -90,8 +96,9 @@ module.exports = {
       },
     },
     {
-      files: ['*.stories.jsx'],
+      files: ['*.stories.{jsx,tsx}'],
       rules: {
+        'no-console': 'off',
```
Comment (dmethvin-gov) [inline .eslintrc.js:101]: Stories can still use console methods.

File: .eslintrc.js [hunk 2]
```diff
@@ -61,6 +61,12 @@ module.exports = {
         'testing-library/render-result-naming-convention': 'warn',
       },
     },
+    {
+      files: ['./packages/**/*.{js,jsx,ts,tsx}'],
+      rules: {
+        'no-console': ['error', { allow: ['warn', 'error'] }],
```
Comment (dmethvin-gov) [inline .eslintrc.js:67]: Shipped code can't have any console method except console.warn and console.error.

File: packages/design-system-tokens/src/index.ts [hunk 3]
```diff
@@ -39,12 +39,12 @@ const tokenExporter = (inputType: string, exportType: string): number => {
   }
 
   const help = (error: string) => {
-    console.log(`\n error: ${error}`);
-    console.log('-------------------------------------------------------------');
-    console.log(' usage : yarn build input_type output_type');
-    console.log(`         where input_type can be ${INPUT_TYPES}`);
-    console.log(`         and output_type can be ${EXPORT_TYPES}`);
-    console.log('-------------------------------------------------------------\n');
+    console.error(`\n error: ${error}`);
```
Comment (dmethvin-gov) [inline packages/design-system-tokens/src/index.ts:42]: These should be `error`.

File: packages/design-system-tokens/src/lib/file.ts [hunk 4]
```diff
@@ -72,6 +72,7 @@ export const getFileDescriptors = (rootPath: string): FileDescriptor[] => {
 export const writeFile = (filename: string, fileData: string): number => {
   try {
     fs.writeFileSync(filename, fileData);
+    // eslint-disable-next-line no-console
     console.log(`:: wrote ${filename}`);
```
Comment (dmethvin-gov) [inline packages/design-system-tokens/src/lib/file.ts:76]: I wasn't sure if this was spurious or not, so I left it and disabled the error. Glad to remove it if that's the right thing.

File: .eslintrc.js (hunk 2 above)
Comment (pwolfert) [reply .eslintrc.js:67]: Yeah, I like that. @zarahzachz, what do you think?

File: packages/design-system-tokens/src/lib/file.ts (hunk 4 above)
Comment (pwolfert) [reply packages/design-system-tokens/src/lib/file.ts:76]: Let's keep it 👍

File: packages/design-system/src/components/MonthPicker/MonthPicker.tsx [hunk 5]
```diff
@@ -99,7 +99,6 @@ export const MonthPicker = (props: MonthPickerProps) => {
     if (props.onChange) {
       props.onChange(event);
     }
-    console.log('isControlled ', isControlled);
```
Comment (pwolfert) [inline packages/design-system/src/components/MonthPicker/MonthPicker.tsx:102]: I was just [about to remove this](https://github.com/CMSgov/design-system/pull/2893)! Your version with a linter rule is better.

**Review by pwolfert — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (pwolfert) [reply .eslintrc.js:101]: Good call

File: .eslintrc.js (hunk 2 above)
Comment (zarahzachz) [reply .eslintrc.js:67]: This seems like a good enhancement

**Comment by pwolfert** [comment]
Jenkins, not a wolf

**Pull request merged**